presto/dbapi.py

```diff
@@ -41,6 +46,93 @@
+        :raises TypeError: if anything but a dict/list/tuple is passed into parameters
+        """
+        if isinstance(parameters, dict):
+            return {k: self._escape_item(v) for k, v in parameters.items()}
+        elif isinstance(parameters, (list, tuple)):
+            return tuple(self._escape_item(x) for x in parameters)
+        else:
+            raise TypeError("Unsupported parameters type. Parameters: " + str(parameters) + " Type: " +
+                            type(parameters) + ".")
+
+    @staticmethod
+    def _escape_string(item):
+        """
+        Escapes string query parameters.
+        1. UTF-decode string if needed (to support old sqlalchemy)
+        2. Escape single quotes by additional single quotes to prevent SQL injection
+        3. Add surrounding single quotes to be inserted into SQL query later
+        The single quote is the only special character to escape in Presto, and it is escaped by another single quote
+        character. Escaping single quote in string parameters protects from semicolon ";" SQL injections, as well as
+        comment "--" injections, as well as injections with dummy filters like "' or 1=1 --".
+
+        :param str item: The string to escape
+        :return: UTF-decoded, single-quote-escaped string, surrounded by additional single quotes
+        """
+        # Need to decode UTF-8 because of old sqlalchemy.
+        # Newer SQLAlchemy checks dialect.supports_unicode_binds before encoding Unicode strings as byte strings.
+        # The old version always encodes Unicode as byte strings, which would otherwise break string formatting here.
+        if isinstance(item, bytes):
+            item = item.decode('utf-8')
+
+        return "'" + item.replace("'", "''") + "'"
+
+    def _escape_sequence(self, sequence):
+        """
+        Escapes iterable collections. Collections containing other collections will be escaped recursively through
+        _escape_item().
+
+        :param Iterable sequence: The sequence to be escaped
+        :return: String containing the escaped given collection, with parentheses to make a proper SQL array
+        """
+        items = map(str, map(self._escape_item, sequence))
+        return '(' + ", ".join(items) + ')'
+
+    def _escape_item(self, item):
+        """
+        Escapes a single item of the parameters.
+        Python's None is transformed to 'NULL' string to be added as NULL to SQL query.
+        Boolean values are transformed to either 'TRUE' or 'FALSE' to be added to SQL query.
+        Numbers returned as they are (they don't require escaping).
+        Strings are escaped in _escape_string().
+        Iterable collections are processed recursively down to individual booleans, numbers or strings.
+
+        :param item: The item to escape
+        :return: An int, a float, or a string
+        """
+        if item is None:
+            return 'NULL'
+        elif isinstance(item, bool):
```

Comment:
`True` and `False` are actually instances of `int` in Python, and they can be embedded into %s and %(named)s parameters via the % operator. I added this `isinstance(item, bool)` branch just for explicity.